### PR TITLE
Implement missing features in vcsim

### PR DESCRIPTION
**Pull request by anfernee**
- Add CustomFieldsManager
- Add Portgroup related operations

### CODE REVIEW

File: vim25/soap/client.go [hunk 1]
```diff
@@ -353,35 +353,39 @@ func (c *Client) SetCertificate(cert tls.Certificate) {
 	// Extension certificate
 	t.TLSClientConfig.Certificates = []tls.Certificate{cert}
 
-	// Proxy to vCenter host on port 80
-	host, _ := splitHostPort(c.u.Host)
+	/*
+		WHY?
```
Comment (anfernee) [inline vim25/soap/client.go]: Not exactly sure why proxy it for client requests with certificate. needs to discuss it with Doug.

File: object/distributed_virtual_switch.go [hunk 2]
```diff
@@ -65,3 +65,15 @@ func (s DistributedVirtualSwitch) AddPortgroup(ctx context.Context, spec []types
 
 	return NewTask(s.Client(), res.Returnval), nil
 }
+
+func (s DistributedVirtualSwitch) FetchDVPorts(ctx context.Context) ([]types.DistributedVirtualPort, error) {
```
Comment (dougm) [inline object/distributed_virtual_switch.go]: Can you update govc/dvs/portgroup/info.go to use this too?

File: simulator/dvs.go [hunk 3]
```diff
@@ -122,3 +123,56 @@ func (s *DistributedVirtualSwitch) ReconfigureDvsTask(req *types.ReconfigureDvs_
...
+			Key:     pg.Key,
+			Config: types.DVPortConfigInfo{
+				Setting: pg.Config.DefaultPortConfig,
+			},
+		})
+
+		if pg.PortKeys == nil {
+			continue
+		}
+		for _, key := range pg.PortKeys {
```
Comment (dougm) [inline simulator/dvs.go]: Not covered in the style guide, but if you look closer at the existing code, blank lines between statements.  It is cosmetic, but I find makes the code easier to read later.  I won't comment on all the places I'd personally insert blank lines in this review, just wanted to make a note of it.

**Comment by anfernee** [comment]
fixed the issues. to be clear, the filter functionality of FetchDVPorts is not fully implemented. I would add it later.

**Pull request merged**